src/backend/main.ts

```diff
@@ -1049,24 +1078,41 @@ ipcMain.handle(
       return false
     })
 
+    powerMonitor.removeListener('suspend', handleSuspend)
+    powerMonitor.removeListener('resume', handleResume)
+
     // Stop display sleep blocker
     if (powerDisplayId !== null) {
       logInfo('Stopping Display Power Saver Blocker', LogPrefix.Backend)
       powerSaveBlocker.stop(powerDisplayId)
     }
 
     // Update playtime and last played date
-    const finishedPlayingDate = new Date()
-    tsStore.set(`${appName}.lastPlayed`, finishedPlayingDate.toISOString())
-    // Playtime of this session in minutes
-    const sessionPlaytime =
-      (finishedPlayingDate.getTime() - startPlayingDate.getTime()) / 1000 / 60
+    currentSession.end = new Date()
+    playTimeSessions.push(currentSession)
+    tsStore.set(`${appName}.lastPlayed`, currentSession.end.toISOString())
+
+    const sessionPlaytime = playTimeSessions.reduce(
+      (acc, next) =>
+        acc + (next.end.getTime() - next.start.getTime()) / 1000 / 60,
+      0
+    )
     const totalPlaytime =
       sessionPlaytime + tsStore.get(`${appName}.totalPlayed`, 0)
     tsStore.set(`${appName}.totalPlayed`, Math.floor(totalPlaytime))
 
-    if (runner === 'gog') {
-      await updateGOGPlaytime(appName, startPlayingDate, finishedPlayingDate)
+    const { disablePlaytimeSync } = GlobalConfig.get().getSettings()
+    if (!disablePlaytimeSync) {
+      for (const session of playTimeSessions) {
+        if (runner === 'gog') {
+          await updateGOGPlaytime(appName, session.start, session.end)
```

Comment:
maybe we don't need the `await` here? just so syncing the playtime doesn't block other actions if for any reason the requests are slow or the update fails

Reply:
I'm not sure how backend would behave when we post sessions in different order, I'd rather keep it in sync